polkadot-api/src/polkadotApi.ts

```diff
@@ -0,0 +1,326 @@
+        const {json, pair} = keyring.addUri(payload.mnemonic, payload.password, {
+          name: payload.name,
+          network: payload.network,
+          isFavorite: payload.isFavorite,
+          isExternal: payload.isExternal,
+        });
+        window.ReactNativeWebView.postMessage(
+          JSON.stringify({
+            type: addAccount.resultType,
+            payload: {account: json},
+          }),
+        );
+
+        if (!pair.isLocked) {
+          pair.lock();
+        }
+        break;
+      }
+
+      case restoreAccount.type: {
+        try {
+          const pair = keyring.restoreAccount(payload.json, payload.password);
+          keyring.saveAccountMeta(pair, {
+            network: payload.network,
+            isExternal: false,
+            isFavorite: payload.isFavorite,
+          });
+          const json = pair.toJson(payload.password);
+          window.ReactNativeWebView.postMessage(
+            JSON.stringify({
+              type: restoreAccount.resultType,
+              payload: {account: json},
+            }),
+          );
+
+          if (!pair.isLocked) {
+            pair.lock();
+          }
+        } catch {
+          window.ReactNativeWebView.postMessage(
+            JSON.stringify({
+              type: restoreAccount.resultType,
+              payload: {isError: true, message: 'Unable to decode using the supplied password.'},
+            }),
+          );
+        }
+        break;
+      }
+
+      case exportAccount.type: {
+        const pair = keyring.getPair(payload.address);
+        try {
+          const json = pair.toJson(payload.password);
+          window.ReactNativeWebView.postMessage(
+            JSON.stringify({
+              type: exportAccount.resultType,
+              payload: {account: json},
+            }),
+          );
+
+          if (!pair.isLocked) {
+            pair.lock();
+          }
+        } catch {
+          window.ReactNativeWebView.postMessage(
+            JSON.stringify({
+              type: exportAccount.resultType,
+              payload: {isError: true, message: 'Unable to decode using the supplied password.'},
+            }),
+          );
+        }
+        break;
+      }
+
+      case addExternalAccount.type: {
+        const {json} = keyring.addExternal(payload.address, {
+          network: payload.network,
+          isFavorite: payload.isFavorite,
+        });
+        window.ReactNativeWebView.postMessage(
+          JSON.stringify({
+            type: addExternalAccount.resultType,
+            payload: {account: json},
+          }),
+        );
+        break;
+      }
+
+      case forgetAccount.type: {
+        keyring.forgetAccount(payload.address);
+        window.ReactNativeWebView.postMessage(
+          JSON.stringify({
+            type: forgetAccount.resultType,
+            payload: {address: payload.address},
+          }),
+        );
+        break;
+      }
+
+      // case 'TOGGLE_FAVORITE': {
```

Comment:
What about this?

Reply:
This was a less priority functionality, updating the account meta data to mark favourite accounts. I have already added this in a separate branch. Will open a separate PR once the clean-up is merged into this.